File: blockchain/pi_network.py
import requests

class PiNetwork:
    def __init__(self, node_url):
        self.node_url = node_url

    def get_balance(self, address):
        url = f"{self.node_url}/balance/{address}"
        response = requests.get(url)
        return response.json()["balance"]

    def send_transaction(self, sender, receiver, amount):
        url = f"{self.node_url}/transaction"
        data = {"sender": sender, "receiver": receiver, "amount": amount}
        response = requests.post(url, json=data)
        return response.json()["tx_hash"]

File: projects/PiWalletBot/server/api/piNetwork.js
import axios from 'axios';

const piNetworkApi = axios.create({
  baseURL: 'https://api.pi.network',
});

export const getPiBalance = async (address) => {
  const response = await piNetworkApi.get(`/balance/${address}`);
  return response.data;
};

export const sendPiTransaction = async (from, to, amount) => {
  const response = await piNetworkApi.post('/transaction', {
    from,
    to,
    amount,
  });
  return response.data;
};
